pkg/api/defaults.go

```diff
@@ -732,6 +749,20 @@ func (p *Properties) setMasterProfileDefaults(isUpgrade bool) {
 	if p.MasterProfile.OSDiskCachingType == "" {
 		p.MasterProfile.OSDiskCachingType = string(compute.CachingTypesReadWrite)
 	}
+
+	if !isUpgrade {
+		if p.MasterProfile.DataDiskSSDType == "" {
+			p.MasterProfile.DataDiskSSDType = Premium
+		}
+		if p.MasterProfile.DataDiskSSDType == Ultra {
+			if p.MasterProfile.UltraSSDEnabled == nil {
```

Comment:
what happens if `UltraSSDEnabled` is false and `DataDiskSSDType` is Ultra?

Reply:
That will be disallowed in validation (// TODO)